framework/web/helpers/CHtml.php

```diff
@@ -2369,7 +2369,9 @@ public static function listData($models,$valueField,$textField,$groupField='')
 	 */
 	public static function value($model,$attribute,$defaultValue=null)
 	{
-		if(is_scalar($attribute) || $attribute===null)
+		if($attribute===null)
```

Comment:
`null` could be a valid key for an array, returning `$defaultValue` blindly is incorrect.